tools/mpremote/mpremote/main.py

```diff
@@ -213,421 +400,102 @@ def usage_error(cmd, exp_args, msg):
+            cmd_parser.add_argument(
+                "next_command", nargs=argparse.REMAINDER, help=f"Next {_PROG} command"
+            )
+            args = cmd_parser.parse_args(command_args)
 
-            if pyb is None:
-                pyb = do_connect(["auto"])
+            # Execute command.
+            handler_func(state, args)
 
-            if need_raw_repl:
-                if not pyb.in_raw_repl:
-                    pyb.enter_raw_repl(soft_reset=auto_soft_reset)
-                    auto_soft_reset = False
-            else:
-                if pyb.in_raw_repl:
-                    pyb.exit_raw_repl()
-            if is_action:
-                did_action = True
-
-            if cmd == "disconnect":
-                do_disconnect(pyb)
-                pyb = None
-                auto_soft_reset = True
-            elif cmd == "soft-reset":
-                pyb.enter_raw_repl(soft_reset=True)
-                auto_soft_reset = False
-            elif cmd == "mount":
-                unsafe_links = False
-                if args[0] == "--unsafe-links" or args[0] == "-l":
-                    args.pop(0)
-                    unsafe_links = True
-                path = args.pop(0)
-                pyb.mount_local(path, unsafe_links=unsafe_links)
-                print(f"Local directory {path} is mounted at /remote")
-            elif cmd == "umount":
-                pyb.umount_local()
-            elif cmd in ("exec", "eval", "run"):
-                follow = True
-                if args[0] == "--no-follow":
-                    args.pop(0)
-                    follow = False
-                if cmd == "exec":
-                    buf = args.pop(0)
-                elif cmd == "eval":
-                    buf = "print(" + args.pop(0) + ")"
-                else:
-                    filename = args.pop(0)
-                    try:
-                        with open(filename, "rb") as f:
-                            buf = f.read()
-                    except OSError:
-                        print(f"{_PROG}: could not read file '{filename}'")
-                        return 1
-                ret = execbuffer(pyb, buf, follow)
-                if ret:
-                    return ret
-            elif cmd == "fs":
-                do_filesystem(pyb, args)
-            elif cmd == "edit":
-                do_edit(pyb, args)
-            elif cmd == "repl":
-                do_repl(pyb, args)
-
-        if not did_action:
-            if pyb is None:
-                pyb = do_connect(["auto"])
-            if pyb.in_raw_repl:
-                pyb.exit_raw_repl()
-            do_repl(pyb, args)
+            # Get any leftover unprocessed args.
+            remaining_args = args.next_command + extra_args
+
+        # If no commands were "actions" then implicitly finish with the REPL.
+        if state.run_repl_on_completion():
+            do_repl(state)
```

Comment:
this call fails because it's defined as `do_repl(state, args)`

Reply:
Accidental cleanup in one of the later updates. Fixed.